bftengine/src/bftengine/DbMetadataStorage.cpp

```diff
@@ -146,6 +146,30 @@ Status DBMetadataStorage::multiDel(const ObjectIdsVector &objectIds) {
   }
   return dbClient_->multiDel(keysVec);
 }
+void DBMetadataStorage::setDontLoadStorageOnStartupFlag() { dontLoadStorageOnStartup = true; }
+void DBMetadataStorage::cleanDB() {
+  // Note that this method is called from the destructor, we don't need to use the mutex
+  // First, Set the number of stored items to 0 such that when the replica startup it won't load the already stored
+  // metadata (see isNewStorage method);
+  uint32_t zero = 0;
+  atomicWrite(objectsNumParameterId_, (char *)&zero, sizeof(zero));
+  LOG_INFO(logger_,
+           "set the number of metadata storage to 0. This was done in order to load a fresh metadata on the next "
+           "replica startup");
+
+  // Second, to clean things up, clear all the records from the DB
+  ObjectIdsVector objectIds;
+  for (const auto &id : objectIdToSizeMap_) {
+    objectIds.push_back(id.first);
+  }
+  multiDel(objectIds);
+}
+
+DBMetadataStorage::~DBMetadataStorage() {
+  if (dontLoadStorageOnStartup) {
```

Comment:
Since exceptions are not allowed to leave destructors, it might be useful to catch exceptions here, log them, and `std::terminate()` the application.

Reply:
done